runner/app/live/streamer/streamer.py

```diff
@@ -142,12 +144,19 @@ async def run_ingress_loop(self, done: Event):
 
                 # crop the max square from the center of the image and scale to 512x512
                 # most models expect this size especially when using tensorrt
-                width, height = frame.size
+                frame_array = np.array(frame)
```

Comment:
Can you avoid doing any processing if resize is not necessary? so maybe keep cropping on pillow, which is cheap, and move the opencv conversion stuff to inside that resize branch. wdyt?

In production setup, the trickle client may already resize the image for us. At some point this should be a pipeline-specific behavior so this logic should be here anyway (and makes simpler to write other clients like the e2e test you're building). So we should avoid doing unnecessary pillow->cv->pillow if no resize is required here.

Reply:
takes less than 1ms that conversion, but good catch!